Without this patch we fail a lot of tests with -DSILENT_NO_TAINT_SUPPORT:
```

Test Summary Report
-------------------
op/taint.t                                                         (Wstat: 65280 (exited 255) Tests: 23 Failed: 17)
  Failed tests:  2-9, 11-18, 23
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 1065 tests but ran 23.
perf/taint.t                                                       (Wstat: 0 Tests: 4 Failed: 2)
  Failed tests:  1, 3
run/runenv.t                                                       (Wstat: 0 Tests: 106 Failed: 1)
  Failed test:  25
run/switcht.t                                                      (Wstat: 0 Tests: 13 Failed: 4)
  Failed tests:  1, 3, 7, 9
../cpan/Encode/t/taint.t                                           (Wstat: 64000 (exited 250) Tests: 498 Failed: 250)
  Failed tests:  1-248, 497-498
  Non-zero exit status: 250
../cpan/Scalar-List-Utils/t/tainted.t                              (Wstat: 768 (exited 3) Tests: 5 Failed: 3)
  Failed tests:  3-5
  Non-zero exit status: 3
../cpan/Test-Harness/t/compat/test-harness-compat.t                (Wstat: 1536 (exited 6) Tests: 120 Failed: 6)
  Failed tests:  6-7, 98-99, 102-103
  Non-zero exit status: 6
../cpan/Text-ParseWords/t/taint.t                                  (Wstat: 0 Tests: 2 Failed: 2)
  Failed tests:  1-2
../cpan/Test-Harness/t/regression.t                                (Wstat: 8704 (exited 34) Tests: 7632 Failed: 34)
  Failed tests:  3023, 3026, 3028, 3031-3037, 3042-3043
                3048-3049, 3054-3056, 3062-3064, 6787, 6790
                6792, 6795, 6798, 6802-6803, 6806-6808
                6816-6817, 6819-6820
  Non-zero exit status: 34
../lib/locale.t                                                    (Wstat: 0 Tests: 681 Failed: 157)
  Failed tests:  17-26, 32-37, 45-49, 57-61, 64-68, 71-75
                78-82, 85-89, 92-96, 98-102, 116-120, 123-128
                131-136, 155, 169, 171-176, 178-183, 185-190
                192-197, 451, 453, 455, 457, 459, 461, 463
                465, 467, 469, 471, 473, 475, 477, 507
                509, 511, 513, 515, 517, 519, 521, 523
                525, 527, 529, 531, 533, 563, 565, 567
                569, 571, 573, 575, 577, 579, 581, 603
                605, 607, 609, 611, 613, 615, 617, 619
                621, 643, 645, 647, 649, 651, 653, 655
                657, 659, 661
../lib/overload.t                                                  (Wstat: 0 Tests: 5363 Failed: 780)
  Failed tests:  622-623, 628, 633, 636, 641, 646, 649-650
                655, 660, 663, 668, 673, 676-677, 682, 687
                690, 695, 700, 703-704, 709, 714, 717, 722
                727, 730-731, 736, 741, 744, 749, 754, 757-758
                763, 768, 771, 776, 781, 784-785, 790, 795
                798, 803, 808, 811-812, 817, 822, 825, 830
                835, 838-839, 844, 849, 852, 857, 862, 865-866
                871, 876, 879, 884, 889, 892-893, 898, 903
                906, 911, 916, 919-920, 925, 930, 933, 938
                943, 946-947, 952, 957, 960, 965, 970, 973-974
                979, 984, 987, 992, 997, 1000-1001, 1006
                1011, 1014, 1019, 1024, 1027-1028, 1033
                1038, 1041, 1046, 1051, 1054-1055, 1060
                1065, 1068, 1073, 1078, 1081-1082, 1087
                1092, 1095, 1100, 1105, 1108-1109, 1114
                1119, 1122, 1127, 1132, 1135-1136, 1141
                1146, 1149, 1154, 1159, 1162-1163, 1168
                1173, 1176, 1181, 1186, 1189-1190, 1195
                1200, 1203, 1208, 1213, 1216-1217, 1222
                1227, 1230, 1235, 1240, 1243-1244, 1249
                1254, 1257, 1262, 1267, 1270-1271, 1276
                1281, 1284, 1289, 1294, 1297-1298, 1303
                1308, 1311, 1316, 1321, 1324-1325, 1330
                1335, 1338, 1343, 1348, 1351-1352, 1357
                1362, 1365, 1370, 1375, 1378-1379, 1384
                1389, 1392, 1397, 1402, 1405-1406, 1411
                1416, 1419, 1424, 1429, 1432-1433, 1438
                1443, 1446, 1451, 1456, 1459-1460, 1465
                1470, 1473, 1478, 1483, 1486-1487, 1492
                1497, 1500, 1505, 1510, 1513-1514, 1519
                1524, 1527, 1532, 1537, 1540-1541, 1546
                1551, 1554, 1559, 1564, 1567-1568, 1573
                1578, 1581, 1586, 1591, 1594-1595, 1600
                1605, 1608, 1613, 1618, 1621-1622, 1627
                1632, 1635, 1640, 1645, 1648-1649, 1654
                1659, 1662, 1667, 1672, 1675-1676, 1681
                1686, 1689, 1694, 1699, 1702-1703, 1708
                1713, 1716, 1721, 1726, 1729-1730, 1735
                1740, 1743, 1748, 1753, 1756-1757, 1762
                1767, 1770, 1775, 1780, 1783-1784, 1789
                1794, 1797, 1802, 1807, 1810-1811, 1816
                1821, 1824, 1829, 1834, 1837-1838, 1843
                1848, 1851, 1856, 1861, 1864-1865, 1870
                1875, 1878, 1883, 1888, 1891-1892, 1897
                1902, 1905, 1910, 1915, 1918-1919, 1924
                1929, 1932, 1937, 1942, 1945-1946, 1951
                1956, 1959, 1964, 1969, 1972-1973, 1978
                1983, 1986, 1991, 1996, 1999-2000, 2005
                2010, 2013, 2018, 2023, 2026-2027, 2032
                2037, 2040, 2045, 2050, 2053-2054, 2059
                2064, 2067, 2072, 2077, 2080-2081, 2086
                2091, 2094, 2099, 2104, 2107-2108, 2113
                2118, 2121, 2126, 2131, 2134-2135, 2140
                2145, 2148, 2153, 2158, 2161-2162, 2167
                2172, 2175, 2180, 2185, 2188-2189, 2194
                2199, 2202, 2207, 2212, 2215-2216, 2221
                2226, 2229, 2234, 2239, 2242-2243, 2248
                2253, 2256, 2261, 2266, 2269-2270, 2275
                2280, 2283, 2288, 2293, 2296-2297, 2302
                2307, 2310, 2315, 2320, 2323-2324, 2329
                2334, 2337, 2342, 2347, 2350-2351, 2356
                2361, 2364, 2369, 2374, 2377-2378, 2383
                2388, 2391, 2396, 2401, 2404-2405, 2410
                2415, 2418, 2423, 2428, 2431-2432, 2437
                2442, 2445, 2450, 2455, 2458-2459, 2464
                2469, 2472, 2477, 2482, 2485-2486, 2491
                2496, 2499, 2504, 2509, 2512-2513, 2518
                2523, 2526, 2531, 2536, 2539-2540, 2545
                2550, 2553, 2558, 2563, 2566-2567, 2572
                2577, 2580, 2585, 2590, 2593-2594, 2599
                2604, 2607, 2612, 2617, 2620-2621, 2626
                2631, 2634, 2639, 2644, 2647-2648, 2653
                2658, 2661, 2666, 2671, 2674-2675, 2680
                2685, 2688, 2693, 2698, 2701-2702, 2707
                2712, 2715, 2720, 2725, 2728-2729, 2734
                2739, 2742, 2747, 2752, 2755-2756, 2761
                2766, 2769, 2774, 2779, 2782-2783, 2788
                2793, 2796, 2801, 2806, 2809-2810, 2815
                2820, 2823, 2828, 2833, 2836-2837, 2842
                2847, 2850, 2855, 2860, 2863-2864, 2869
                2874, 2877, 2882, 2887, 3538-3539, 3544
                3549, 3552, 3557, 3562, 3565-3566, 3571
                3576, 3579, 3584, 3589, 3592-3593, 3598
                3603, 3606, 3611, 3616, 3619-3620, 3625
                3630, 3633, 3638, 3643, 3646-3647, 3652
                3657, 3660, 3665, 3670, 3673-3674, 3679
                3684, 3687, 3692, 3697, 3700-3701, 3706
                3711, 3714, 3719, 3724, 3727-3728, 3733
                3738, 3741, 3746, 3751, 3754-3755, 3760
                3765, 3768, 3773, 3778, 3781-3782, 3787
                3792, 3795, 3800, 3805, 3808-3809, 3814
                3819, 3822, 3827, 3832, 3835-3836, 3841
                3846, 3849, 3854, 3859, 3862-3863, 3868
                3873, 3876, 3881, 3886, 3889-3890, 3895
                3900, 3903, 3908, 3913, 3916-3917, 3922
                3927, 3930, 3935, 3940, 3943-3944, 3949
                3954, 3957, 3962, 3967, 3970-3971, 3976
                3981, 3984, 3989, 3994, 3997-3998, 4003
                4008, 4011, 4016, 4021, 4024-4025, 4030
                4035, 4038, 4043, 4048, 4051-4052, 4057
                4062, 4065, 4070, 4075, 4078-4079, 4084
                4089, 4092, 4097, 4102, 4105-4106, 4111
                4116, 4119, 4124, 4129, 4186-4187, 4192
                4197, 4200, 4205, 4210, 4213-4214, 4219
                4224, 4227, 4232, 4237, 4789, 4803, 4845
                4859, 4873, 4887, 4892-4893, 4898, 4903
                4906, 4911-4912, 4917, 4922, 4925, 4930-4931
                4936, 4941, 4944-4945, 4950, 4955
../lib/warnings.t                                                  (Wstat: 0 Tests: 941 Failed: 3)
  Failed tests:  767-769
../lib/perl5db.t                                                   (Wstat: 0 Tests: 142 Failed: 1)
  Failed test:  18
Files=2755, Tests=1129357, 352 wallclock secs (224.21 usr 14.80 sys + 1880.75 cusr 75.86 csys = 2195.62 CPU)
Result: FAIL
Finished test run at Thu Mar 30 17:05:33 2023.
make: *** [makefile:865: test_harness] Error 236
```

--- a/.github/workflows/testsuite.yml
+++ b/.github/workflows/testsuite.yml
@@ -263,6 +263,7 @@ jobs:
           - "-Duserelocatableinc"
           - "-Dcc='clang'"
           - "-Dcc='g++'"
+          - "-Accflags=-DSILENT_NO_TAINT_SUPPORT"
 
     steps:
       - name: Install System dependencies

--- a/configpm
+++ b/configpm
@@ -576,6 +576,8 @@ $_ = <<'!END!';
 EOT
 #proper lexicographical order of the keys
 my %seen_var;
+my @v_define = ( "taint_support=''\n",
+                 "taint_disabled=''\n" );
 $heavy_txt .= join('',
     map { $_->[-1] }
     sort {$a->[0] cmp $b->[0] }
@@ -583,7 +585,7 @@ $heavy_txt .= join('',
     map {
         /^([^=]+)/ ? [ $1, $_ ]
                    : [ $_, $_ ] # shouldnt happen
-    } @v_others, @v_forced
+    } (@v_others, @v_forced, @v_define)
 ) . "!END!\n";
 
 # Only need the dynamic byteorder code in Config.pm if 'byteorder' is one of
@@ -593,6 +595,32 @@ if ($Common{byteorder}) {
 } else {
     $heavy_txt .= $byteorder_code;
 }
+
+$heavy_txt .= <<'EOT';
+s/(byteorder=)(['"]).*?\2/$1$2$Config::byteorder$2/m;
+
+EOT
+
+$heavy_txt .= <<'EOF_TAINT_INIT';
+{
+    # We have to set this up late as Win32 does not build miniperl
+    # with the same defines and CC flags as it builds perl itself.
+    my $defines = join " ", (Internals::V)[0,1];
+    if (
+        $defines =~ /\b(SILENT_NO_TAINT_SUPPORT)\b/ ||
+        $defines =~ /\b(NO_TAINT_SUPPORT)\b/
+    ){
+        my $which = $1;
+        my $taint_disabled = ($which eq "SILENT_NO_TAINT_SUPPORT")
+                             ? "silent" : "define";
+        s/^(taint_disabled=['"])(["'])/$1$taint_disabled$2/m;
+    }
+    else {
+        my $taint_support = 'define';
+        s/^(taint_support=['"])(["'])/$1$taint_support$2/m;
+    }
+}
+EOF_TAINT_INIT
 
 if (@need_relocation) {
 $heavy_txt .= 'foreach my $what (qw(' . join (' ', @need_relocation) .
@@ -612,8 +640,6 @@ EOT
 }
 
 $heavy_txt .= <<'EOT';
-s/(byteorder=)(['"]).*?\2/$1$2$Config::byteorder$2/m;
-
 my $config_sh_len = length $_;
 
 our $Config_SH_expanded = "\n$_" . << 'EOVIRTUAL';
@@ -1014,29 +1040,21 @@ ENDOFTAIL
 if ($Opts{glossary}) {
   open(GLOS, '<', $Glossary) or die "Can't open $Glossary: $!";
 }
-my %seen = ();
 my $text = 0;
 $/ = '';
 my $errors= 0;
 
+my %glossary;
+
+my $fc;
+my $item;
+
 sub process {
   if (s/\A(\w*)\s+\(([\w.]+)\):\s*\n(\t?)/=item C<$1>\n\nFrom F<$2>:\n\n/m) {
-    my $c = substr $1, 0, 1;
-    unless ($seen{$c}++) {
-      print CONFIG_POD <<EOF if $text;
-=back
-
-EOF
-      print CONFIG_POD <<EOF;
-=head2 $c
-
-=over 4
-
-EOF
-     $text = 1;
-    }
+    $item = $1;
+    $fc = substr $item, 0, 1;
   }
-  elsif (!$text || !/\A\t/) {
+  elsif (!$item || !/\A\t/) {
     warn "Expected a Configure variable header",
       ($text ? " or another paragraph of description" : () ),
       ", instead we got:\n$_";
@@ -1068,6 +1086,7 @@ EOF
   s/(?<![.<\'\"])\b([A-Z_]{2,})\b(?![\'\"])/C<$1>/g;	# UNISTD
   s/(?<![.<\'\"])\b(?!the\b)(\w+)\s+macro\b/C<$1> macro/g; # FILE_cnt macro
   s/n[\0]t/n't/g;		# undo can't, won't damage
+  $glossary{$fc}{$item} .= $_;
 }
 
 if ($Opts{glossary}) {
@@ -1075,7 +1094,6 @@ if ($Opts{glossary}) {
     <GLOS>;				# Skip the preamble
   while (<GLOS>) {
     process;
-    print CONFIG_POD;
   }
   if ($errors) {
     die "Errors encountered while processing $Glossary. ",
@@ -1086,9 +1104,43 @@ if ($Opts{glossary}) {
   }
 }
 
-print CONFIG_POD <<'ENDOFTAIL';
+$glossary{t}{taint_support} //= <<EOF_TEXT;
+=item C<taint_support>
 
-=back
+From define: C<SILENT_NO_TAINT_SUPPORT> or C<NO_TAINT_SUPPORT>
+
+If this perl is compiled with support for taint mode this variable will
+be set to 'define', if it is not it will be set to the empty string.
+Either of the above defines will result in it being empty.  This property
+was added in version 5.37.11. See also L</taint_disabled>.
+
+EOF_TEXT
+
+$glossary{t}{taint_disabled} //= <<EOF_TEXT;
+=item C<taint_disabled>
+
+From define: C<SILENT_NO_TAINT_SUPPORT> or C<NO_TAINT_SUPPORT>
+
+If this perl is compiled with support for taint mode this variable will
+be set to the empty string, if it was compiled with
+C<SILENT_NO_TAINT_SUPPORT> defined then it will be set to be "silent",
+and if it was compiled with C<NO_TAINT_SUPPORT> defined it will be
+'define'. Either of the above defines will results in it being a true
+value. This property was added in 5.37.11. See also L</taint_support>.
+
+EOF_TEXT
+
+if ($Opts{glossary}) {
+    foreach my $fc (sort keys %glossary) {
+        print CONFIG_POD "=head2 $fc\n\n=over 4\n\n";
+        foreach my $item (sort keys %{$glossary{$fc}}) {
+            print CONFIG_POD $glossary{$fc}{$item};
+        }
+        print CONFIG_POD "=back\n\n";
+    }
+}
+
+print CONFIG_POD <<'ENDOFTAIL';
 
 =head1 GIT DATA
 

--- a/ext/XS-APItest/t/call.t
+++ b/ext/XS-APItest/t/call.t
@@ -14,7 +14,7 @@ BEGIN {
     plan(538);
     use_ok('XS::APItest')
 };
-
+use Config;
 #########################
 
 # f(): general test sub to be called by call_sv() etc.
@@ -343,8 +343,11 @@ for my $fn_type (qw(eval_pv eval_sv call_sv)) {
 # DAPM 9-Aug-04. A taint test in eval_sv() could die after setting up
 # a new jump level but before pushing an eval context, leading to
 # stack corruption
+SKIP: {
+    skip("Your perl was built without taint support", 1)
+        unless $Config{taint_support};
 
-fresh_perl_is(<<'EOF', "x=2", { switches => ['-T', '-I../../lib'] }, 'eval_sv() taint');
+    fresh_perl_is(<<'EOF', "x=2", { switches => ['-T', '-I../../lib'] }, 'eval_sv() taint');
 use XS::APItest;
 
 my $x = 0;
@@ -357,4 +360,4 @@ sub f {
 eval { my @a = sort f 2, 1;  $x++};
 print "x=$x\n";
 EOF
-
+}

--- a/lib/B/Deparse.t
+++ b/lib/B/Deparse.t
@@ -307,14 +307,19 @@ x(); z()
 .
 EOCODH
 
-is runperl(stderr => 1, switches => [ '-MO=-qq,Deparse', $path, '-T' ],
+SKIP: {
+    skip("Your perl was built without taint support", 1)
+        unless $Config::Config{taint_support};
+
+    is runperl(stderr => 1, switches => [ '-MO=-qq,Deparse', $path, '-T' ],
            prog => "format =\n\@\n\$;\n.\n"),
-   <<'EOCODM', '$; on format line';
-format STDOUT =
-@
-$;
-.
-EOCODM
+        <<~'EOCODM', '$; on format line';
+        format STDOUT =
+        @
+        $;
+        .
+        EOCODM
+}
 
 is runperl(stderr => 1, switches => [ '-MO=-qq,Deparse,-l', $path ],
            prog => "format =\n\@\n\$foo\n.\n"),
@@ -537,10 +542,14 @@ is runperl(stderr => 1, switches => [ '-MO=-qq,Deparse', $path ],
    "sub BEGIN {\n    \$main::{'f'} = \\!0;\n}\n",
    '&PL_sv_yes constant (used to croak)';
 
-is runperl(stderr => 1, switches => [ '-MO=-qq,Deparse', $path, '-T' ],
+SKIP: {
+    skip("Your perl was built without taint support", 1)
+        unless $Config::Config{taint_support};
+    is runperl(stderr => 1, switches => [ '-MO=-qq,Deparse', $path, '-T' ],
            prog => '$x =~ (1?/$a/:0)'),
-  '$x =~ ($_ =~ /$a/);'."\n",
-  '$foo =~ <branch-folded match> under taint mode';
+        '$x =~ ($_ =~ /$a/);'."\n",
+        '$foo =~ <branch-folded match> under taint mode';
+}
 
 unlike runperl(stderr => 1, switches => [ '-MO=-qq,Deparse', $path, '-w' ],
                prog => 'BEGIN { undef &foo }'),

--- a/lib/Config.t
+++ b/lib/Config.t
@@ -51,6 +51,39 @@ ok( exists $Config{d_fork},  "has d_fork");
 
 ok(!exists $Config{d_bork},  "has no d_bork");
 
+{
+    # check taint_support and tain_disabled are set up as expected.
+
+    ok( exists $Config{taint_support}, "has taint_support");
+
+    ok( exists $Config{taint_disabled}, "has taint_disabled");
+
+    is( $Config{taint_support}, ($Config{taint_disabled} ? "" : "define"),
+        "taint_support = !taint_disabled");
+
+    ok( ($Config{taint_support} eq "" or $Config{taint_support} eq "define"),
+        "taint_support is a valid value");
+
+    ok( ( $Config{taint_disabled} eq "" or $Config{taint_disabled} eq "silent" or
+        $Config{taint_disabled} eq "define"),
+        "taint_disabled is a valid value");
+
+    my @opts = Config::non_bincompat_options();
+    my @want_taint_disabled = ("", "define", "silent");
+    my @want_taint_support = ("define", "", "");
+    my ($silent_no_taint_support) = grep $_ eq "SILENT_NO_TAINT_SUPPORT", @opts;
+    my ($no_taint_support) = grep $_ eq "NO_TAINT_SUPPORT", @opts;
+    my $no_taint_support_count = 0 + grep /NO_TAINT_SUPPORT/, @opts;
+    my $want_count = $silent_no_taint_support ? 2 : $no_taint_support ? 1 : 0;
+
+    is ($no_taint_support_count, $want_count,
+        "non_bincompat_options info on taint support is as expected");
+    is( $Config{taint_disabled}, $want_taint_disabled[$no_taint_support_count],
+        "taint_disabled is aligned with non_bincompat_options() data");
+    is( $Config{taint_support}, $want_taint_support[$no_taint_support_count],
+        "taint_support is aligned with non_bincompat_options() data");
+}
+
 like($Config{ivsize}, qr/^(4|8)$/, "ivsize is 4 or 8 (it is $Config{ivsize})");
 
 # byteorder is virtual, but it has rules.

--- a/t/TEST
+++ b/t/TEST
@@ -522,6 +522,36 @@ sub dump_tests {
     exit(0);
 }
 
+sub filter_taint_tests {
+    my $tests = shift;
+    require Config;
+    return unless $Config::Config{taint_disabled} eq "define";
+
+    # These are test files which are known to fail with -DNO_TAINT_SUPPORT
+    # but which do not have "taint" in their name, nor have shebang lines
+    # with -t or -T in them. So we exclude them specifically instead.
+    my %known_tainter = map { $_ => 0 } (
+        '../cpan/Test-Harness/t/regression.t',
+        '../cpan/Test-Harness/t/source_handler.t',
+        '../cpan/Test-Harness/t/compat/inc-propagation.t',
+    );
+    @$tests = grep {
+        my $file = $_;
+        open my $ifh, "<", $file
+            or die "Failed to read: '$file': $!";
+        my $line = <$ifh>;
+        my $keep = $file=~/taint/ ? 0 : ($known_tainter{$file} // 1);
+        if ($line=~/^#!.*perl\s+-(\w+)/) {
+            my $switch = $1;
+            if ($switch =~ s/[Tt]//) {
+                $keep = 0;
+            }
+        }
+        $keep
+    } @$tests;
+}
+
+
 unless (@ARGV) {
     # base first, as TEST bails out if that can't run
     # then comp, to validate that require works
@@ -607,6 +637,8 @@ unless (@ARGV) {
 };
 
 dump_tests(\@ARGV) if $dump_tests;
+
+filter_taint_tests(\@ARGV);
 
 if ($::deparse) {
     _testprogs('deparse', '',   @ARGV);

--- a/t/harness
+++ b/t/harness
@@ -433,6 +433,8 @@ for (@tests) {
 
 dump_tests(\@tests) if $dump_tests;
 
+filter_taint_tests(\@tests);
+
 my %options;
 
 my $type = 'perl';

--- a/t/run/switchDx.t
+++ b/t/run/switchDx.t
@@ -48,9 +48,14 @@ END {
     fresh_perl_like("print qq(hello)", qr/define raw/,
                     { stderr => 1, switches => [ '-Di' ] },
                    "-Di defaults to stderr");
-    fresh_perl_like("print qq(hello)", qr/define raw/,
+    SKIP: {
+        skip("Your perl was built without taint support", 1)
+            unless $Config{taint_support};
+
+        fresh_perl_like("print qq(hello)", qr/define raw/,
                     { stderr => 1, switches => [ '-TDi' ] },
                    "Perlio debug output to STDERR with -TDi (no PERLIO_DEBUG)");
+    }
 }
 {
     # -DXv tests

--- a/t/test.pl
+++ b/t/test.pl
@@ -1327,6 +1327,13 @@ sub run_multiple_progs {
         my $dummy;
         ($dummy, @prgs) = _setup_one_file(shift);
     }
+    my $taint_disabled;
+    if (! eval {require Config; 1}) {
+        warn "test.pl had problems loading Config: $@";
+        $taint_disabled = '';
+    } else {
+        $taint_disabled = $Config::Config{taint_disabled};
+    }
 
     my $tmpfile = tempfile();
 
@@ -1378,6 +1385,10 @@ sub run_multiple_progs {
     } elsif (defined $file) {
         $name = "test from $file at line $line";
     }
+
+        if ($switch=~/[Tt]/ and $taint_disabled eq "define") {
+            $reason{skip} ||= "This perl does not support taint";
+        }
 
 	if ($reason{skip}) {
 	SKIP:

